raiden-ts/src/channels/epics.ts

```diff
@@ -999,13 +999,13 @@ export const channelUpdateEpic = (
     getTokenNetworkContract,
     config$,
   }: RaidenEpicDeps,
-): Observable<channelSettle.failure> =>
+): Observable<never> =>
   action$.pipe(
     filter(isActionOf(channelClose.success)),
     filter((action) => !!action.payload.confirmed),
-    // wait 2 newBlock actions go through after channelClose confirmation, to ensure any pending
+    // wait a newBlock go through after channelClose confirmation, to ensure any pending
```

Comment:
No.. this actually gets confirmed on block before the action, I'm just giving a little more time

Reply:
But why?

Reply:
Sorry, actually it gets confirmed on the same block as action. We could technically actually send the `update` the channel on the first unconfirmed `channelClose.success` event, the blockchain itself doesn't care about confirmations. We already are waiting for this event to get confirmed (`filter` above), so this `delayWhen` is just to give one more block to it in case we're actually fetching **past events** from the blockchain, and a syncronous confirmed `channelSettle.success` could be coming right after this. Look at `channelEventsEpic` to check the past fetching of events is synchronous.

Reply:
And also, if no `channelSettle.success` comes, the new block could be the one to set the channel.state as `settleable`, which also marks when we can't update it anymore, so this _waiting one block_ is enough, and the tests check this.

Reply:
Hmm. Sounds a bit fragile to me. But I trust you there. Else I get things never finished. :sweat_smile: